### ISSUE TITLE
Script Canvas Nodes: The ‘On Tick’ and ‘On Entity Activated’ nodes Behavior.

### ISSUE THREAD

**Issue by G13D** [o3de/o3de#17297]
**Describe the bug**
The Script Canvas Execute nodes ‘On Tick’ and ‘On Entity Activated’ are not functioning if they are later disconnected from the ‘Connect’ slots by another executable graph node.

It was an experience with the first version of the O3DE Multiplayer Sample that I downloaded from GitHub, It was a single script canvas file and when I opened it to just play around, the debug nodes were not functioning. Previously, I had accidentally connected a new ‘On Tick’ node to them.

Before connecting the node slot to the second node connection slot, they were working and printing simple text
![image](https://github.com/o3de/o3de/assets/58872135/48c014b0-ed48-4f6a-b7d2-c0adf9273c22)

Entered game mode
(Script) - ------  Print  On Entity Activated  [49361168290348]
(Script) - ------  Print On Graph
(Script) - ------  Print On Tick  0.0257

1.
The event handlers are now connected and the graph is saved.
![image](https://github.com/o3de/o3de/assets/58872135/141c8009-f09b-4aa7-9eab-a22253b26d65)
2.
After the editor restarts, the connected nodes are functioning. But now, we will disconnect the nodes’ connections and save the graph again.

Entered game mode
(Script) - ------  Print On Graph
(Script) - ------  Print  On Entity Activated  [14554674193637]
(Script) - ------  Print On Tick  0.0409
3.
Disconnect the slots and save the graph. Enter game mode with CTRL + G. The ‘On Tick’ and ‘On Entity Activated’ nodes are not functioning to print anything.

![image](https://github.com/o3de/o3de/assets/58872135/625e891e-60d7-4784-af7c-c1992519f5e2)
Entered game mode
(Script) - ------  Print On Graph
4.
Now, if the ‘On Tick’ node is copied to another empty graph and you press save, there will be an error: ‘Asset Processor Status’ shows a red flag - ‘FAILED’

12/20/2023 6:23 PM | ERROR | ScriptCanvasBuilder: Graph conversion to abstract code model failed: * This graph defines no functions, it is never activated, and will never execute. Add a Start node or connect an event handler or define functions. from AssetBuilder
![image](https://github.com/o3de/o3de/assets/58872135/6787fbae-70f6-409f-847e-649a8af2a75f)

**Expected behavior**
Does self-execution occur if there is no connection logic on them?

**Actual behavior**
It was confusing why a lot of other connected nodes were not working, etc.

**Found in Branch**
Release 23.10.1 and Nightly Development

**Desktop/Device (please complete the following information):**
 - Device: [PC] 
 - OS: [Windows]
 - Version [10]
 - CPU [I7]

**Additional context**

**Comment by lsemp3d** [comment]
I've been unable to reproduce this on the latest development branch.

Not using `Connect`
```
Entered game mode
(Script) - -- On Entity Activated [28298841149659]
(Script) - -- On Graph Start
(Script) - >> On Tick: 0.0866
Exited game mode
```

Using `Connect`

```
Entered game mode
(Script) - -- On Entity Activated [29707590422747]
(Script) - -- On Graph Start
(Script) - OnTick Connected
(Script) - >> On Tick: 0.2565
Exited game mode
```

Disconnected `Connect`, saved, then Ctrl+G

```
Entered game mode
(Script) - -- On Entity Activated [30626713424091]
(Script) - -- On Graph Start
(Script) - >> On Tick: 0.0862
Exited game mode
```

Could there be any other steps? are you able to try to reproduce it on the latest development branch to verify?

**Comment by G13D** [comment]
> I've been unable to reproduce this on the latest development branch.
> 
> Not using `Connect`
> 
> ```
> Entered game mode
> (Script) - -- On Entity Activated [28298841149659]
> (Script) - -- On Graph Start
> (Script) - >> On Tick: 0.0866
> Exited game mode
> ```
> 
> Using `Connect`
> 
> ```
> Entered game mode
> (Script) - -- On Entity Activated [29707590422747]
> (Script) - -- On Graph Start
> (Script) - OnTick Connected
> (Script) - >> On Tick: 0.2565
> Exited game mode
> ```
> 
> Disconnected `Connect`, saved, then Ctrl+G
> 
> ```
> Entered game mode
> (Script) - -- On Entity Activated [30626713424091]
> (Script) - -- On Graph Start
> (Script) - >> On Tick: 0.0862
> Exited game mode
> ```
> 
> Could there be any other steps? are you able to try to reproduce it on the latest development branch to verify?

Yes, there are several grammatical errors and the sentences could be more clear. Here’s a corrected version:

The slots should be connected and the script saved, When the engine launches next time, disconnect the slots, save, and test on tick and on entity activated. It should not print anything or stop printing if you enter game mode once two times Depends on Something. This is what I observed for myself.

I have installed latest Installer and built from source. The new projects in development  same behavior

[OnTickOnEntityActivated_Backup.scriptcanvas.json](https://github.com/o3de/o3de/files/13824452/OnTickOnEntityActivated_Backup.scriptcanvas.json)
[OnTickOnEntityActivated.scriptcanvas.json](https://github.com/o3de/o3de/files/13824453/OnTickOnEntityActivated.scriptcanvas.json) Broken
Asset Processor: 

Info | origin: AssetProcessor | Unable to remove file Cache/pc/scriptcanvas/ontickonentityactivated_vm.luac to copy source file user/AssetProcessorTemp/JobTemp-MMVUBN/OnTickOnEntityActivated_VM.luac in... (We may retry)

Info | origin: AssetProcessor | SUCCESS:  after failure, we later succeeded to copy/move file E:/O3DE/.../Cache/pc/scriptcanvas/ontickonentityactivated_vm.luac

Info | origin: AssetProcessor | Unable to remove file /Cache/pc/scriptcanvas/ontickonentityactivated.scriptcanvas_compiled to copy source file /user/AssetProcessorTemp/JobTemp-fvffBF/OnTickOnEntityActivated.scriptcanvas_compiled in... (We may retry)

Info | origin: AssetProcessor | SUCCESS:  after failure, we later succeeded to copy/move file E:/O3DE/   /Cache/pc/scriptcanvas/ontickonentityactivated.scriptcanvas_compiled

 Info | origin: AssetProcessor | Unable to remove file Cache/pc/scriptcanvas/ontickonentityactivated.scriptcanvas_fn_compiled to copy source file user/AssetProcessorTemp/JobTemp-fvffBF/OnTickOnEntityActivated.scriptcanvas_fn_compiled in... (We may retry)

**Comment by G13D** [comment]
The slots should be connected and the script saved, When the engine launches next time, disconnect the slots, save, and test on tick and on entity activated. It should not print anything or stop printing if you enter game mode once two times Depends on Something. This is what I observed for myself.

I have installed latest Installer and built from source. The new projects in development  same behavior

[OnTickOnEntityActivated_Backup.scriptcanvas.json](https://github.com/o3de/o3de/files/13824452/OnTickOnEntityActivated_Backup.scriptcanvas.json)
[OnTickOnEntityActivated.scriptcanvas.json](https://github.com/o3de/o3de/files/13824453/OnTickOnEntityActivated.scriptcanvas.json) Broken
 Asset Processor: 
```json
info | origin: AssetProcessor | Unable to remove file Cache/pc/scriptcanvas/ontickonentityactivated_vm.luac to copy source file user/AssetProcessorTemp/JobTemp-MMVUBN/OnTickOnEntityActivated_VM.luac in... (We may retry)

Info | origin: AssetProcessor | SUCCESS:  after failure, we later succeeded to copy/move file E:/O3DE/.../Cache/pc/scriptcanvas/ontickonentityactivated_vm.luac

Info | origin: AssetProcessor | Unable to remove file /Cache/pc/scriptcanvas/ontickonentityactivated.scriptcanvas_compiled to copy source file /user/AssetProcessorTemp/JobTemp-fvffBF/OnTickOnEntityActivated.scriptcanvas_compiled in... (We may retry)

Info | origin: AssetProcessor | SUCCESS:  after failure, we later succeeded to copy/move file E:/O3DE/   /Cache/pc/scriptcanvas/ontickonentityactivated.scriptcanvas_compiled

 Info | origin: AssetProcessor | Unable to remove file Cache/pc/scriptcanvas/ontickonentityactivated.scriptcanvas_fn_compiled to copy source file user/AssetProcessorTemp/JobTemp-fvffBF/OnTickOnEntityActivated.scriptcanvas_fn_compiled in... (We may retry) ```

**Comment by G13D** [comment]
https://github.com/o3de/o3de/assets/58872135/8eaa9b88-c14a-4020-9dff-b7ccd0f79f36

Some Nodes Shows UI of JSON Error, When I Want to Save it and Most of the Nodes Are of the "Get Type" and Have an Additional Input Slot of Brown

**Comment by G13D** [comment]
![image](https://github.com/o3de/o3de/assets/58872135/2025c054-2386-4453-9791-08b970ffe849)
[json_Error_On_Listed_Nodes02.scriptcanvas.json](https://github.com/o3de/o3de/files/13903278/json_Error_On_Listed_Nodes02.scriptcanvas.json)
I have not checked others

**Comment by lsemp3d** [comment]
Thank you for the additional details! I'll investigate and try to reproduce the problem, doing a diff of the files you included I do see an entry that is likely causing the problem:

```
"m_autoConnectToGraphOwner": false
```

I'm not sure what's causing that to be added but I'll check
